ts/Data/DataSeriesConverter.ts

```diff
@@ -0,0 +1,267 @@
+            yValueId = '_' + series.id;
+            pointArrayMap = series.pointArrayMap || ['y'];
+            pointArrayMapLength = pointArrayMap.length;
+            options = series.options;
+            keys = options.keys;
+            data = series.options.data || [];
+
+            for (let j = 0, jEnd = data.length; j < jEnd; j++) {
+                elem = data[j];
+                y = columnMap.y ? columnMap.y + yValueId : 'y' + yValueId;
+                columns = {};
+                needsArrayMap = pointArrayMapLength > 1;
+
+                if (typeof elem === 'number') {
+                    columns[y] = elem;
+                    columns.x = j;
+                } else if (elem instanceof Array) {
+                    xIndex = keys && keys.indexOf('x') > -1 ? keys.indexOf('x') : 0;
+                    yIndex = keys && keys.indexOf('y') > -1 ? keys.indexOf('y') : 1;
+
+                    if (needsArrayMap) {
+                        for (let k = 0; k < pointArrayMapLength; k++) {
+                            yValueIndex = keys && keys.indexOf(pointArrayMap[k]) > -1 ?
+                                keys.indexOf(pointArrayMap[k]) : k + elem.length - pointArrayMapLength;
+
+                            yValueName = columnMap[pointArrayMap[k]] ?
+                                columnMap[pointArrayMap[k]] : pointArrayMap[k];
+                            columns[yValueName + yValueId] = elem[yValueIndex];
+                        }
+                    } else {
+                        columns[y] = elem[yIndex];
+                    }
+
+                    columns.x = elem.length - pointArrayMapLength > 0 ? elem[xIndex] : j;
+
+                } else if (elem instanceof Object) {
+                    if (needsArrayMap) {
+                        const elemSet = elem as Record<string, DataTableRow.CellType>;
+
+                        for (let k = 0; k < pointArrayMapLength; k++) {
+                            yValueName = columnMap[pointArrayMap[k]] ?
+                                columnMap[pointArrayMap[k]] : pointArrayMap[k];
+                            columns[yValueName + yValueId] = elemSet[yValueName];
+                        }
+                    } else {
+                        columns[y] = elem.y;
+                    }
+
+                    columns.x = elem.x || j;
+                }
+
+                id = '' + columns.x;
+                row = table.getRow(id);
+
+                if (!row) {
+                    columns.id = id;
+                    row = new DataTableRow(columns);
+                    table.insertRow(row, eventDetail);
+                } else if (columns[y]) {
+                    row.insertCell(y, columns[y], eventDetail);
+                }
+            }
+        }
+
+        return table;
+    }
+}
+/* *
+ *
+ *  Namespace
+ *
+ * */
+namespace DataSeriesConverter {
+    export interface Options {
+        type?: string;
+        columnMap?: Record<string, string>;
```

Comment:
One usage difficulty with `columnMap` is, that you can not map a column index. So you can not simply convert `Series.parallelArrays` to a map, where the column index is equal to the array index.